### DISCUSSION TITLE
Weather Seems inaccurate

### DISCUSSION THREAD

**Discussion by mij2** [riffnshred/nhl-led-scoreboard#263] [General]
I've tried putting in the location a few different ways (bigger town near me, zip code, actual town, etc.) and something seems off about the weather displays. I'm not sure if there's a conversion issue (imperial units) or if I'm not fully understanding what the weather and forecast boards are displaying.
![IMG_20210403_101907](https://user-images.githubusercontent.com/81831002/113488825-1728a400-9486-11eb-8327-048ba865e3ad.jpg)
![IMG_20210403_101853](https://user-images.githubusercontent.com/81831002/113488827-198afe00-9486-11eb-9cac-845e45bf1ba0.jpg)

For additional context, the weather is currently 67F, with a low of 39F, feels like 66F. Tomorrow is supposed to be about 75F and a low of 51F. The pictures are from a couple of hours ago, but still you can see there is some sort of disconnect.

I'm new to the project and would gladly help dig into it. I have some moderate programming/querying experience and a lot of general technology experience.

**Comment by pjockey** [comment]
Which weather supplier are you using?

**Comment by mij2** [comment]
Here are the weather settings I have. The wxalert is using NWS.

![image](https://user-images.githubusercontent.com/81831002/113498845-4f57d300-94d6-11eb-9042-b43884657d9b.png)

**Reply by pjockey** [reply]
And you've tried to change the data_feed from EC to NWS?. What do you have in the "location"?

**Reply by mij2** [reply]
I have not, but I think that's because when I've run the ./nhl_setup, it only gives the option of EC or OWM. I can manually insert NWS to see if that solves the issue.

**Reply by pjockey** [reply]
at the top of the script, place your city in there as will.

**Reply by mij2** [reply]
I do have the city in there already. I had also tried zip code.

**Reply by mij2** [reply]
Replacing that line seems to cause an error with the weather. When setting up the config through nhl_setup, the only options there are EC and OWM. I believe it's only the weather alerts that allows you to use NWS.

I ran the software again while viewing the logs in console, but it looks like it just doesn't see the "NWS" under `weather` as valid even those the config passes validation.
![image](https://user-images.githubusercontent.com/81831002/113527783-360d6000-9584-11eb-958a-ee1100ba1e07.png)
![image](https://user-images.githubusercontent.com/81831002/113527796-49203000-9584-11eb-9ee7-eb9d70927e49.png)


Here is my config for the weather board for the above logs:
![image](https://user-images.githubusercontent.com/81831002/113527866-81c00980-9584-11eb-84e5-06aa1b7e01d4.png)

**Reply by pjockey** [reply]
Here is my weather. There are some difference, but not enough, I would think, to cause it not to work

```
"weather": {
            "enabled": true,
            "view": "full",
            "units": "metric",
            "duration": 60,
            "data_feed": "EC",
            "owm_apikey": "",
            "update_freq": 5,
            "show_on_clock": true,
            "forecast_enabled": true,
            "forecast_days": 3,
            "forecast_update": 1
        },
```
Did you install the noaa_sdk?  pip install noaa-sdk

**Reply by mij2** [reply]
What else would I have to setup for NOAA to be used for weather? Doing a little bit of poking around, I think maybe the fact that I didn't set up OWM (wasn't sure the type of API key needed), that it just takes the closest location to me for data from EC.

I guess I might have to request a different change, but it seems like the weather board needs to be updated to use NWS. I'm very new to this project, linux, and rpi in general, but not new to programming. I could potentially help with this, but would likely need a little bit of pointing in the right direction.

**Reply by pjockey** [reply]
you can go here, https://openweathermap.org/api and sign up for an account and get the API. I believe it is free for personal use and if one does not pull too many requests during a day. 
I read somewhere, the wiki or the set up instructions, that the noaa-sdk needed to be installed for NWS and perhaps another package, but can't seem to locate it at the moment.

**Reply by mij2** [reply]
Thank you very much!! Once I added that API key and waited a few minutes, it seems to have picked up the weather and it appears to be close enough to what I'm seeing in my weather app. I'm new to github, but I think I'd still recommend potentially using the NWS/NOAA info for weather and forecast as well, but this works for the time being.

**Reply by falkyre** [reply]
I wrote the weather code and the issue with NOAA is the station observations are too inaccurate and don't update as much as what OWM provides (the stations ranged from conditions being updated every hour to every 12 hours).  I also didn't find something that worked the way I wanted it to to provide current observations (to start) and forecasts.  Originally all of the weather was done with the Dark Sky APi but then Apple closed it down and I had to rewrite the whole thing from scratch.  The NWS feed is used for alerts because the OWM doesn't provide alerts. EC is Environment Canada and should only be used if you are in a Canadian city. That feed provides forecast, current obs and alerts. 

I feel that the best observations and forecasts come from the OWM provider and don't see a need to add NOAA.  And the bonus of using OWM is that it's global so it can be used by users outside of the United States (which we do have some in Europe using the board).

**Reply by mij2** [reply]
That makes way more sense. You did a good job with it. I was just unsure about the OWM API Key initially and had hoped there would be a way to do it without signing up for one. Ultimately, it ended up being a solid solution. Thank you for all the hard work!

**Comment by pjockey** [comment]
[README.txt](https://github.com/riffnshred/nhl-led-scoreboard/files/6256124/README.txt)
